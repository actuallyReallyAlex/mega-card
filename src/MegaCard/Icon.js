import React from 'react'
import { Box } from 'grommet'
import { colors, icons, propTypes } from '../constants'

const Icon = ({ color, icon }) => {
  const Figure = icons[icon]
  return (
    <Box>
      <Box
        align="center"
        height="42px"
        justify="center"
        round="full"
        style={{ backgroundColor: colors[color] }}
        width="42px"
      >
        <Figure color="white" size="28px" />
      </Box>
    </Box>
  )
}

Icon.propTypes = {
  color: propTypes.color.isRequired,
  icon: propTypes.icon.isRequired
}

export default Icon
