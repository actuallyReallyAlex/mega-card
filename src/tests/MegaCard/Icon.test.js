import React from 'react'
import Icon from '../../MegaCard/Icon'
import { shallow } from 'enzyme'

const defaultProps = {
  color: 'blue',
  icon: 'Cube'
}

describe('<Icon />', () => {
  afterEach(() => {
    // defaultProps.handler.mockClear()
  })

  test('Should render the <Icon /> component.', () => {
    const component = shallow(<Icon {...defaultProps} />)
    expect(component).toMatchSnapshot()
  })
})
