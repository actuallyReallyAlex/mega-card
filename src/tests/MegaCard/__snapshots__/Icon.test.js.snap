// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> Should render the <Icon /> component. 1`] = `ShallowWrapper {}`;
